packages/vite/src/node/utils.ts

```diff
@@ -38,8 +38,10 @@ export const normalizeId = (id: string): string =>
   id.replace(/(\s*>\s*)/g, ' > ')
 
 export function isBuiltin(id: string): boolean {
-  const deepMatch = id.match(deepImportRE)
-  id = deepMatch ? deepMatch[1] || deepMatch[2] : id
+  const builtins = [...builtinModules]
+  if (!builtins.includes('fs/promises')) builtins.push('fs/promises')
+  if (!builtins.includes('diagnostics_channel')) builtins.push('diagnostics_channel') 
```

Comment:
```suggestion
  if (!builtins.includes('diagnostics_channel')) {
    builtins.push('diagnostics_channel')
  }
```

... Or see my other review ...